src/analysis/souffle/examples/multimic/multimic-preferences.authlogic

```diff
@@ -0,0 +1,335 @@
+//  describes how this user would like the policy framework to handle its own 
+//  data. The three preferences are:
+//     "permissive" -- the user permits anything to be done with their data
+//     "only_store_transcript" -- only transcripts can be stored, not audio
+//     "only_notify" -- the only thing that can be done with audio is notify 
+//     there is a microphone
+//  - setsPrivacyResolution(<type_of_data>, <resolution_suggestion>) gives the 
+//  device owner a way to decide how disputes among user policies are handled
+//  (cases where not everyone's preference can be met). The types of
+//  resolution are:
+//     "meet_requests" -- only allow actions everyone has consented to
+//     "stay_on" -- the microphone must stay on even if this goes agaist the 
+//     preferences of some users in the room
+
+"UserA" says {
+  ownsTag("UserA", "UserA_userSpeech").
+  ownsTag("UserA", "UserA_speechTranscript").
+
+  // UserA expresses that its preference about how audio is handled is 
+  // permissive. Any action can be done with it and it can be transformed in 
+  // any way.
+  "UserA" hasPrivacyPreference("audio", "permissive").
+  // UserA expresses that transcription and storage of transcripts should stay 
+  // on even if this conflicts with the requests of others.
+  "UserA" setsPrivacyResolution("audio", "stay_on").
+
+  // User A gives the policy principal the right to decide that any action can 
+  // be done with UserA's data and to transform this data in any way.
+  "AmbientPrivacyPolicy" canSay PrincipalX may(ActionX, TagX) :-
+    isAction(ActionX), isPrincipal(PrincipalX), isTag(TagX).
+  "AmbientPrivacyPolicy" canSay downgrades(PathX, TagX) :-
+    isPath(PathX), isTag(TagX).
+  "AmbientPrivacyPolicy" canSay hasTag(PathX, TagX) :-
+    isPath(PathX), isTag(TagX).
+
+  // This is stuff that needs to be generated behind the scenes
+  // (Clearly A's policy should have no knowledge of this)
+  isAction("deliver_privacy_notifications").
+  isAction("local_computation").
+  isAction("store").
+  isPrincipal("AmbientPrivacyPolicy").
+  isPrincipal("Notifier").
+  isPrincipal("MicCompute").
+  isPrincipal("SpeechToTextStorageSink").
+  isPath("audioIn").
+  isPath("asrStorageConn").
+  isPath("speechToTextIn").
+  isTag("UserA_userSpeech").
+  isTag("UserA_speechTranscript").
+}
+
+// UserB is a guest user. UserB is somewhat privacy sensitive. They want their
+// voice recorded only as a transcript and will only recognize authorities from
+// companies that they trust.
+"UserB" says {
+  ownsTag("UserB", "UserB_userSpeech").
+  ownsTag("UserB", "UserB_speechTranscript").
+
+  // UserB's preference for handling audio is to only store transcripts of what 
+  // was said, but not raw audio.
+  "UserB" hasPrivacyPreference("audio", "only_store_transcript").
+  // UserB says that disputes should be handled by meeting the requests of all 
+  // people in the room.
+  "UserA" setsPrivacyResolution("audio", "meet_requests").
```

Comment:
Sorry this is just a typo, it should say `UserB`.

Actually, this statement that B is making about UserA will not be believed by `AmbientPrivacyPolicy`, because it only trusts users to state their own `setsPrivacyResolution` (via the rule on line 165). So if this typo happened in the wild, it wouldn't lead to a bad access decision.

Reply:
Fixed the typo now.